ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -444,4 +491,89 @@ private void terminateQuery(final TerminateQuery terminateQuery) throws Exceptio
         new CommandStatus(CommandStatus.Status.TERMINATED, "Query terminated")
     );
   }
+
+  private void terminateCluster(
+      final List<String> keepSources,
```

Comment:
why do we need both `keepSources` and `deleteSources`? only 1 branch of the if statement below will ever execute. Is it for convenience. Anyway, if we have both then you shouldn't be able to specify both properties as if you do then only the `deleteSources` branch will execute.

Reply:
I refactored and now using one list along with a type parameter indicating the type of the list that can only bee `KEEP` or `DELETE`.